components/os/windows.go

```diff
@@ -62,37 +62,62 @@ func (*Windows) GetRunAgentCmd(parameters string) string {
 	return `& "$env:ProgramFiles\Datadog\Datadog Agent\bin\agent.exe" ` + parameters
 }
 
-func getAgentURL(version AgentVersion) (string, error) {
-	minor := strings.ReplaceAll(version.Minor, "~", "-")
-	fullVersion := fmt.Sprintf("%v.%v", version.Major, minor)
-	if version.BetaChannel {
-		finder, err := newAgentURLFinder("https://s3.amazonaws.com/dd-agent-mstesting/builds/beta/installers_v2.json")
-		if err != nil {
-			return "", err
-		}
+func getWindowsRepositoryURL(version AgentVersion) string {
+	baseURL := "https://ddagent-windows-stable.s3.amazonaws.com"
 
-		url, err := finder.findVersion(fullVersion)
-		if err != nil {
-			// Try to handle custom build
-			minor = strings.TrimSuffix(minor, "-1")
-			return fmt.Sprintf("https://s3.amazonaws.com/dd-agent-mstesting/builds/beta/ddagent-cli-%v.%v.msi", version.Major, minor), nil
-		}
+	if version.Repository == TrialRepository {
+		baseURL = "https://ddagent-windows-trial.s3.amazonaws.com"
+	}
+	if version.Repository == StagingRepository {
+		baseURL = "https://dd-agent-mstesting.s3.amazonaws.com/builds"
+	}
 
-		return url, nil
+	// In the prod / trial repositories, the stable channel is at the root of the repository,
+	// not prefixed by the channel name
+	if (version.Repository == ProdRepository || version.Repository == TrialRepository) && version.Channel == StableChannel {
+		return baseURL
 	}
 
-	finder, err := newAgentURLFinder("https://ddagent-windows-stable.s3.amazonaws.com/installers_v2.json")
+	return fmt.Sprintf("%v/%v", baseURL, version.Channel)
+}
+
+func getAgentURL(version AgentVersion) (string, error) {
+	// TODO: using the testing repository on Windows will require more work
+	// - there is no installers_v2.json file to use
+	// - the pipeline ID is not enough to find the exact filename to use, as it is structured like:
+	// datadog-agent-<last tag>.git.<number of commits since tag>.<commit short sha>.pipeline.<pipeline id>-1-x86_64.msi
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Thank you for the todo !

Reply:
do you mean this installers_v2.json ?
https://s3.amazonaws.com/ddagent-windows-stable/installers_v2.json

Reply:
Yes. The `installers_v2.json` file is currently not created in the temporary repositories used by the kitchen tests (and now e2e tests) in the `datadog-agent` pipeline.